### PR TITLE
Issue #1966 Case Sensitive method

### PR DESCRIPTION
**Pull request by gregw**
#1966 
Modified the compliance violations to warn if case insensitivity is applied to a header

This is a not a fix, it is simple a change to start warning of compliance violations if a method is converted to uppercase.

Signed-off-by: Greg Wilkins <gregw@webtide.com>

### CODE REVIEW

File: jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java [hunk 1]
```diff
@@ -648,8 +648,26 @@ private boolean parseLine(ByteBuffer buffer)
                         _length=_string.length();
                         _methodString=takeString();
                         HttpMethod method=HttpMethod.CACHE.get(_methodString);
-                        if (method!=null)
-                            _methodString=legacyString(_methodString,method.asString());
+                        if (method!=null && method.asString()!=_methodString)
+                        {
+                            // TODO See #1966 Case Sensitive Method
+                            if (method.asString()!=_methodString)
```
Comment (sbordet) [inline jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java]: I don't think this is right. 

First of all, it's the same check done in the outer if, so this line is always `true`. 

Then, it's comparing `String`s with `!=`; `methodString` is always generated from what is being parsed via `takeString()`, so it is always `true` that `method.asString() != _methodString`, no ?

Comment (gregw) [reply jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java]: oops - I moved the condition to the outer if, so I could get rid on the inner if... but forgot to remove it. Fixed now.

Comment (sbordet) [reply jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java]: Still not convinced by `String` comparisons.

File: jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java [hunk 2]
```diff
@@ -1271,7 +1271,7 @@ else if (_endOfContent == EndOfContent.UNKNOWN_CONTENT)
                         break;
                     }
                     
-                    if (b==HttpTokens.LINE_FEED && !complianceViolation(RFC2616,"https://tools.ietf.org/html/rfc2616#section-4.2 No colon"))
+                    if (b==HttpTokens.LINE_FEED && !complianceViolation(RFC7230,"https://tools.ietf.org/html/rfc2616#section-4.2 No colon"))
```
Comment (sbordet) [inline jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java]: The `complianceViolation()` call uses RFC7230, but the link points to RFC2616.

Comment (gregw) [reply jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java]: How did you do this comment?  It only appears here and not in the "Files Changed" tab? 
I was in two minds... RFC2616 actually also enforces No Colon, but we didn't check for it then. Strictly speaking this should be legacy, but I think we should keep behaviour.   Stand by...

**Comment by gregw** [comment]
@joakime OK to merge?

**Pull request merged**